### ISSUE TITLE
Autosave issue is back? (from v3.2.0)

### ISSUE THREAD

**Issue by lparolari** [TheElectronWill/night-config#43]
The following code doesn't work from v3.2.0.
If I switch to v3.1.0 it works.

```java
public class ConfigAdapter {

    private File configFile;
    private FileConfig config;

    /**
     * @param configFile File from which read previous preference
     *                    and on which write updated preference
     */
    public ConfigAdapter(@NotNull File configFile) {
        Log.debug("constr:: config file: " + configFile.getAbsolutePath());

        this.configFile = configFile;
        this.config = FileConfig.builder(configFile,JsonFormat.fancyInstance())
                .sync()
                .autosave()
                .autoreload()
                .build();

        ConversionTable conversionTable = new ConversionTable();
        conversionTable.put(Long.class, Long::intValue);
        config = conversionTable.wrapRead(config);
        Log.debug("constr:: build conversion table");

        config.load();
        Log.debug("constr:: config.load()");
    }

    /**
     * Find and return value at searched key, or default.
     *
     * @param k   The key to search
     * @param def The default value if object not found
     * @param <T> The value's type
     * @return The value at {@code k} or {@code def}
     */
    public <T> T get(String k, T def) {
        Log.debug("get:: (" + k + ", " + def + ")");

        T x = config.get(k);
        return x != null ? x : def;
    }

    /**
     * Replaces old value with new one, return old.
     *
     * @param k     The value's key
     * @param v     The value to set
     * @param <T>   The type of old value
     * @return      The old value if any, or null
     */
    public <T> T put(String k, T v) {
        Log.debug("put:: (" + k + ", " + v + ")");

        T x = config.set(k,v);
        return x;
    }

}
```

Tell me if you need some specific information or if I'm doing something wrong.
Thanks!

**Comment by TheElectronWill** [comment]
This is weird because almost nothing changed between 3.1.0 and 3.2.0. See [github diff](https://github.com/TheElectronWill/Night-Config/compare/3.1.1...3.2.0) - look for FileConfig,  FileConfigBuilder, WriteSyncFileConfig, WriteAsyncFileConfig, AutosaveFileConfig and AutoreloadFileConfig.

When you say that it doesn't work, could you explain the result you wanted and what you got instead?
Also, could you [try the code with the latest master](https://jitpack.io/#TheElectronWill/Night-Config/master-SNAPSHOT) commit? Some bugs have been fixed since 3.2.0.

**Comment by lparolari** [comment]
Sorry for the incomplete issue posted yesterday, I was in a hurry.

However, I was using the code above for this operations:
1. reading an empty configuration with a default value of zero;
2. (doing some operations not with config library)
3. updating previously readed config with a value

Example:
1. read database version (0 at the first time)
2. creating database with migrations
3. saving database version (version 5)

As you can see above, the config object was built like this
```java
this.config = FileConfig.builder(configFile,JsonFormat.fancyInstance())
                .sync()
                .autosave()
                .autoreload()
                .build();
```
so the `get` and `put` methods reload and save configurations automatically.

However, at the end of this operations **nothing is updated to the config file**.
It is strange because I have the following lines:
```java
Log.debug("Db:: curr version before: " + currentVer());
upgrade();
Log.debug("Db:: curr version after: " + currentVer());
```
and 
```java
public Integer currentVer() {
    return configuration.get(Keys.DB_VERSION,0);
}
```
The first debug line prints 0, as expected.
The second debug line **prints 5**, as expected, but the **value is not updated** in the file from the version 3.2.0 while in the 3.1.0 this works right.

**Comment by TheElectronWill** [comment]
I've released v3.3.0 with the fix I've mentioned earlier, should be easier to try than the master commit :)
Also, it's available on Maven Central now!
